cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -111,12 +115,18 @@ public void setUp() throws Exception {
       doClear(configuredNamespace, false);
     }
     registeredNamespaces.put(configuredNamespace, deleteUponTeardown);
+    applicationManagers.clear();
     LOG.info("Completed setUp.");
   }
 
   @After
   public void tearDown() throws Exception {
     LOG.info("Beginning tearDown.");
+
+    // Stop any running programs created in @Test
+    for (ApplicationManager manager: applicationManagers) {
```

Comment:
"he notification to persist that KILLED run record tries to write itself to the store."
This causes the next test case to fail? I didn't understand that part.